### PR TITLE
GET /entities/<entity_id>/entitysets Endpoint

### PR DESCRIPTION
**Pull request by mynameisfiber**
This adds a new endpoint attached to entities in order to find all EntitySets which reference the entity. The entitysets can be filtered based on type, collection_id, label or the judgement on the entity within the set.

### CODE REVIEW

File: aleph/views/entities_api.py [hunk 1]
```diff
@@ -576,3 +577,83 @@ def expand(entity_id):
...
+      - Profiles
+    """
+    entity = get_index_entity(entity_id, request.authz.READ)
+
+    parser = QueryParser(request.args, request.authz)
+    collection_ids = [
+        cid
+        for cid in parser.filters.get("collection_id", [])
+        if request.authz.can(cid, request.authz.READ)
+    ]
```
Comment (pudo) [inline aleph/views/entities_api.py:649]: If I read this correctly, then having no `filter:collection_id` filters will return an empty array, and in `.by_entity_id` that will disable the authorisation filter.

File: aleph/views/entities_api.py [hunk 2]
```diff
@@ -576,3 +577,83 @@ def expand(entity_id):
...
+    judgements = parser.filters.get("judgement")
+    labels = parser.filters.get("label")
+    types = parser.filters.get("type")
+
+    entitysets = get_entitysets_by_entity(
```
Comment (pudo) [inline aleph/views/entities_api.py:654]: Maybe we can inline the query building to here? It's only 2, 3 lines and we can avoid one round of indirection.

**Review by pudo — CHANGES_REQUESTED** [review]
I think there's a security thing here.

File: aleph/views/entities_api.py (hunk 2 above)
Comment (mynameisfiber) [reply aleph/views/entities_api.py:654]: Sure thing!

File: aleph/views/entities_api.py (hunk 1 above)
Comment (mynameisfiber) [reply aleph/views/entities_api.py:649]: Correct, but then `EntitySetItemsQuery` will filter out based on the `collection_id`. However, at that point the data is already fetched, so I can also add a:

```
if not collection_ids:
    collection_ids = request.authz.collections(request.authz.READ)
```

Comment (mynameisfiber) [reply aleph/views/entities_api.py:649]: Actually, hold up. I'm confusing this part of code with what I was looking at for #1429. `EntitySetItemsQuery` has no bearing on this.

Comment (pudo) [reply aleph/views/entities_api.py:649]: Yeah I think that would be cool. I don't think `EntitySetItemsQuery` is involved?

Comment (mynameisfiber) [reply aleph/views/entities_api.py:649]: For #1429 I was making sure to myself that `GET /entitysets/XXX/entities` wouldn't leak which lead my down a rabbithole.

**Pull request merged**